src/routers/AppRouter.tsx

```diff
@@ -1,71 +1,88 @@
-/** Libraries */
-import { Routes, Route, BrowserRouter } from "react-router-dom";
-
-import Backdrop from "@mui/material/Backdrop";
-import CircularProgress from "@mui/material/CircularProgress";
-
-/** Middlewares */
-import { PrivateRoute } from "./PrivateRoute";
-import { PublicRoute } from "./PublicRoute";
-
-/** Components */
-import { HomePage, LoginPage } from "../pages";
-import { useAuthStore } from "../hooks";
-import { useEffect } from "react";
-
-
-export const AppRouter = (): JSX.Element => {
-
-  const {
-    _id,
-    checking,
-    startChecking,
-  } = useAuthStore();
-
-  useEffect(() => {
-    startChecking();
-  }, []);
-
-
-  if (checking) {
-    return (
-      <Backdrop
-        sx={{ color: "#fff", zIndex: (theme) => theme.zIndex.drawer + 1 }}
-        open={true}
-      >
-        <CircularProgress
-          color="primary"
-          size="80px"
-          sx={{ display: "block" }}
-        />
-      </Backdrop>
-    );
-  }
-
-  return (
-    <BrowserRouter>
-
-      <Routes>
-
-        <Route
-          path="/"
-          element={
-            <PrivateRoute isAutenticated={!!_id}>
-              <HomePage />
-            </PrivateRoute>
-          }
-        />
-
-        <Route
-          path="login"
-          element={
-            <PublicRoute isAutenticated={!!_id}>
-              <LoginPage />
-            </PublicRoute>
-          }
-        />
-
-      </Routes>
-    </BrowserRouter>
-  );
-};
+/** Libraries */
+import { useEffect, useState } from "react";
+import { Routes, Route, BrowserRouter } from "react-router-dom";
+
+import Backdrop from "@mui/material/Backdrop";
+import CircularProgress from "@mui/material/CircularProgress";
+
+/** Middlewares */
+import { PrivateRoute } from "./PrivateRoute";
+import { PublicRoute } from "./PublicRoute";
+
+/** Pages */
+import { HomePage, LoginPage } from "../pages";
+
+/** Components */
+import { LikeNotification } from "../components/ui/LikeNotification";
+
+/** Custom hooks */
+import { useAuthStore } from "../hooks";
+
+export const AppRouter = (): JSX.Element => {
+  const { _id, checking, startChecking } = useAuthStore();
+
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+
+  /** This is to implement the functionality of
+   * load products when the user cames until the
+   * bottom of the page.
+   */
+  window.addEventListener("scroll", function () {
+    if (
+      window.scrollY + window.innerHeight >=
+      document.documentElement.scrollHeight
+    ) {
+      miFuncion();
+    }
+  });
+
+  function miFuncion() {

```

Comment:
the function name should be changed to something more meaningful like onScrollUpdate

Reply:
Thanks bro I see your comment just today, in the next pull request I will do the change, now I will merge this to try deploy the app